### PR TITLE
wu2021rethinking_saikm200022

### PR DESCRIPTION
**Pull request by saikm200022**

### CODE REVIEW

**Comment by saikm200022** [comment]
@ywen666

File: _summaries/wu2021rethinking_2.md [hunk 1]
```diff
@@ -1,19 +1,71 @@
...
-* How is it realized (technically)?
-* How well does the paper perform?
-* What interesting variants are explored?
+# **Summary - Rethinking "Batch" in BatchNorm, Wu, Johnnson; 2021**
+
+## What is the core idea?
+
+BatchNorm is a commonly used component of neural networks that has greatly increased performance in image classifcation tasks. However, there are certain inherent aspects of BatchNorm that can lead to sub-optimal performance if not addressed correctly. This paper identifies issues primarily around population statistics computation,  training and test set inconsistency, domain shift, and information leakage. The paper also suggests solutions to combat each of these issues to make the most out of BatchNorm in networks.
+
+![Alt Text](wu2021_1f.PNG)
```
Comment (ywen666) [inline _summaries/wu2021rethinking_2.md:16]: I wonder if possible to change the formula to latex, it looks better than the image,

$$y=\frac{x-\mu}{\sqrt{\sigma^2+\epsilon}}$$

File: _summaries/wu2021rethinking_2.md [hunk 2]
```diff
@@ -1,19 +1,71 @@
...
+
+
+## How is it realized (technically)?
+
+There were 4 main experiments performed in this paper:
+
+1. Train a ResNet-50 model on 100 epochs with a mini-batch size = 8192. EMA (Exponential Moving Average) is used to calculate population statistics with lambda = 0.9. Randomly checked and compared the population mean and EMA mean of a random channel in a random BatchNorm Layer. They run a second experiment on a ResNet-50 model with PreciseBN as the population statistic computation method.
```
Comment (ywen666) [inline _summaries/wu2021rethinking_2.md:23]: Nit: on 100 epochs -> for 100 epochs

File: _summaries/wu2021rethinking_2.md [hunk 3]
```diff
@@ -1,19 +1,71 @@
...
+
+## How well does the paper perform?
+
+### **Population Statistic Computation - More Representative Population Statistics**
+
+The paper found that EMA was not successful in finding representative population or mini-batch statistics. As shown  with the discrepancy in the below graph:
+
+![Alt Text](wu2021_1a.PNG)
+
+The paper's suggestion for more precise population statistics was using PreciseBN, which aims to approximate the population statistics using a fixed model state and collecting batch statistics into population statistics. They found that PreciseBN led to better overall accuracy than EMA as it was able to produce population statistics that were more representative of the mini-batch statistics.
```
Comment (ywen666) [inline _summaries/wu2021rethinking_2.md:42]: Maybe explain what is preciseBN here?

File: _summaries/wu2021rethinking_2.md [hunk 4]
```diff
@@ -1,19 +1,71 @@
...
+4. Used the second stage of an R-CNN object detector that takes region-of-interets for each image and derives prediction for that ROI. BatchNorm is used in R-CNN where regions from all images are merged into one mini-batch. This model was implemented with a pre-trained ResNet-50. They tested information leakage with 2 scenarios: using SyncBN and shuffling ROI among GPUs prior to the second stage. 
+
+
+## How well does the paper perform?
+
+### **Population Statistic Computation - More Representative Population Statistics**
+
+The paper found that EMA was not successful in finding representative population or mini-batch statistics. As shown  with the discrepancy in the below graph:
+
+![Alt Text](wu2021_1a.PNG)
```
Comment (ywen666) [inline _summaries/wu2021rethinking_2.md:40]: Any potential reason mentioned in the paper why the discrepancy happens?

File: _summaries/wu2021rethinking_2.md [hunk 5]
```diff
@@ -1,19 +1,71 @@
...
+![Alt Text](wu2021_1f.PNG)
+
+
+## How is it realized (technically)?
+
+There were 4 main experiments performed in this paper:
+
+1. Train a ResNet-50 model on 100 epochs with a mini-batch size = 8192. EMA (Exponential Moving Average) is used to calculate population statistics with lambda = 0.9. Randomly checked and compared the population mean and EMA mean of a random channel in a random BatchNorm Layer. They run a second experiment on a ResNet-50 model with PreciseBN as the population statistic computation method. 
+
+![Alt Text](wu2021_1b.PNG)
```
Comment (ywen666) [inline _summaries/wu2021rethinking_2.md:25]: Same as above, can be written in latex $$$$?

File: _summaries/wu2021rethinking_2.md (hunk 4 above)
Comment (saikm200022) [reply _summaries/wu2021rethinking_2.md:46]: It was, I have added the reasoning to the summary.

File: _summaries/wu2021rethinking_2.md (hunk 3 above)
Comment (saikm200022) [reply _summaries/wu2021rethinking_2.md]: Done.

**Pull request merged**